components/script/webdriver_handlers.rs

```diff
@@ -281,6 +282,24 @@ pub fn handle_get_active_element(
         .unwrap();
 }
 
+pub fn handle_get_page_source(
+    documents: &Documents,
+    pipeline: PipelineId,
+    reply: IpcSender<Result<String, ()>>,
+) {
+    reply
+        .send(documents.find_document(pipeline).ok_or(()).and_then(|doc| {
+            match doc.GetDocumentElement() {
+                Some(elem) => match elem.serialize(TraversalScope::IncludeNode) {
```

Comment:
We should rely on the existing serialization support code by invoking `GetOuterHTML()`, and then fall back to  XMLDocument.serializeToString per [the spec](https://w3c.github.io/webdriver/#get-page-source).